generator/internal/predicate_utils.h

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef GOOGLE_CLOUD_CPP_GENERATOR_INTERNAL_PREDICATE_UTILS_H
+#define GOOGLE_CLOUD_CPP_GENERATOR_INTERNAL_PREDICATE_UTILS_H
+
+#include "google/cloud/optional.h"
+#include <google/protobuf/descriptor.h>
+#include <functional>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+
+/**
+ * Determines if the given method meets the criteria for pagintation.
+ *
+ * https://google.aip.dev/client-libraries/4233
+ */
+bool IsPaginated(google::protobuf::MethodDescriptor const* method);
```

Comment:
(optional) General comment: I think you should consider switching to const-refs instead of pointers all over (other than where you're overriding a method that requires a pointer). The grpc and proto style of using pointers (a choice made ~20 years ago) is no longer a best practice. Unless we _must_ use a pointer, or want _pointer semantics_ (allow nullptr, disallow binding to temporaries, etc.), I think `const&` is a better choice for non-optional input parameters than is `const*`. Using const-refs would be more consistent with our existing code, and our current style guides. Consistency with grpc and protobuf APIs is a non-goal.